### PR TITLE
#474, removing NullValueMapping and refactoring

### PR DESCRIPTION
**Pull request by sjaakd**

### CODE REVIEW

File: core-common/src/main/java/org/mapstruct/BeanMapping.java [hunk 1]
```diff
@@ -27,7 +27,8 @@
 /**
  * Configures the mapping between two bean types.
  * <p>
- * Either {@link #resultType()} or {@link #qualifiedBy()} must be specified.
+ * Either {@link #resultType()} , {@link #qualifiedBy()} or {@link #qualifiedBy()}must be specified.
```
Comment (agudian) [inline core-common/src/main/java/org/mapstruct/BeanMapping.java]: the third is mean to be #nullValueMappingStrategy, right?

**Comment by agudian** [comment]
I didn't find much to complain about... :+1: 

So you can go ahead with the merge (after fixing this one javadoc c/p error). I'll try to finish up the doc-updates for the mapper inheritance in the meantime... :-/

File: core-common/src/main/java/org/mapstruct/BeanMapping.java (hunk 1 above)
Comment (sjaakd) [inline core-common/src/main/java/org/mapstruct/BeanMapping.java]: Yep.. copy paste error.

**Comment by sjaakd** [comment]
> So you can go ahead with the merge (after fixing this one javadoc c/p error). I'll try to finish up the doc-updates for the mapper inheritance in the meantime... :-/

any chance that you tag-along the nullvaluemapping stuff (removing the annotation iso the bean mapping iterate / map mapping )? :smile:

**Comment by agudian** [comment]
> any chance that you tag-along the nullvaluemapping stuff (removing the annotation iso the bean mapping iterate / map mapping )? :smile:

Yes, it's ok - I'll do it along the way.

**Comment by sjaakd** [comment]
> Yes, it's ok - I'll do it along the way.

Thanks!

**Pull request merged**